### ISSUE TITLE
Any suggestion to generate 1-D vectors using diffusion model

### ISSUE THREAD

**Issue by animebing** [lucidrains/denoising-diffusion-pytorch#48]
Recently I am trying to use the diffusion model to do 1-D vector generation task, such as to generate sentence embedding which is originally generated from Bert, I have some questions about it

1. the diffusion model is mainly used to generate images recently, is it possible to use that to generate 1-D vector?
2. if it is possible, which part do I should modify to do the training?
3. if it is impossible, what is the problem?

thanks for your suggestions

**Comment by XDUWQ** [comment]
For previous diffusion models with 1D data, such as spatio-temporal sequences and 1D vector graphs, I think you need to at least modify the channel of Unet.

**Comment by animebing** [comment]
spatial-temporal sequences should be 2D vectors instead of 1D vectors, and there is no time dimension or size dimension in the 1D vectors I am talking here

**Comment by sxl324** [comment]
Hi, have you figured out a way to do 1-D vector training and generation?

**Comment by fikry102** [comment]
> Hi, have you figured out a way to do 1-D vector training and generation?

Is there any solution for this problem?
In my case , each data is a 240-dim vector, [x1,y1,x2,y2, ..., x_N,y_N]

**Comment by lucidrains** [comment]
it is here, in the readme https://github.com/lucidrains/denoising-diffusion-pytorch#1d-sequence

**Comment by fikry102** [comment]
> it is here, in the readme https://github.com/lucidrains/denoising-diffusion-pytorch#1d-sequence

However, I can't understand some number in the code. 
`training_seq = torch.rand(64, 32, 128)`
```
model = Unet1D(
    dim = 64,
    dim_mults = (1, 2, 4, 8),
    channels = 32
)
```
I have 1w samples, each of which is a 240-dim vector, [x1,y1,x2,y2, ..., x_N,y_N].

I don't know how to modify the the channels of Unet1d and have no idea about the specific meaning of "64, 32, 128" for  training_seq.

**Comment by lucidrains** [comment]
@fikry102 i recommend joining a beginner machine learning community online
